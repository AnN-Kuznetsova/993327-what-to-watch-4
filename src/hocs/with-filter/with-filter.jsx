import React, {PureComponent} from 'react';
import {FilterItem} from '../../components/filter-item/filter-item';


export const withFilter = (Component, filterType) => {
  class WithFilter extends PureComponent {
    constructor(props) {
      super(props);

      this.state = {
        filterType,
        activeFilter: null,
      };

      this.renderFilter = this.renderFilter.bind(this);
      this.handleFilterItemClick = this.handleFilterItemClick.bind(this);
    }

    handleFilterItemClick() {}

    renderFilter(filterNames) {
      return (
        <ul className="catalog__genres-list">
          {
            filterNames.map((filterName, index) =>
              <FilterItem
                key={filterName + index}
                filterName={filterName}
                onClick={this.handleFilterItemClick}
              />
            )
          }
        </ul>
      );
    }

    render() {
      return (
        <Component
          {...this.props}
          renderFilter={this.renderFilter}
        />
      );
    }
  }


  WithFilter.propTypes = {};


  return WithFilter;
};
