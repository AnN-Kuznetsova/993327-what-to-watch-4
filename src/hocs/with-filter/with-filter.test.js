import React from "react";
import renderer from "react-test-renderer";

import {withFilter} from "./with-filter";


const Component = (props) => {
  return (
    <div {...props} />
  );
};

const ComponentWithFilter = withFilter(Component);


describe(`Render withFilter`, () => {
  it(`Should match with snapshot`, () => {
    const componentWithFilterSnapshot = renderer.create(
        <ComponentWithFilter />
    ).toJSON();

    expect(componentWithFilterSnapshot).toMatchSnapshot();
  });
});
