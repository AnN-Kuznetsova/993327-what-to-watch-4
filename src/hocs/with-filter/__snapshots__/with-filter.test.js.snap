// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render withFilter Should match with snapshot 1`] = `
<div
  renderFilter={[Function]}
/>
`;
